pytest_httpserver/blocking_http_server.py

```diff
@@ -0,0 +1,163 @@
+from queue import Empty
+from queue import Queue
+from typing import Any
+from typing import Mapping
+from typing import Optional
+from typing import Pattern
+from typing import Union
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+from pytest_httpserver.httpserver import METHOD_ALL
+from pytest_httpserver.httpserver import UNDEFINED
+from pytest_httpserver.httpserver import HeaderValueMatcher
+from pytest_httpserver.httpserver import HTTPServer
+from pytest_httpserver.httpserver import QueryMatcher
+from pytest_httpserver.httpserver import RequestHandler
+from pytest_httpserver.httpserver import URIPattern
+
+
+class BlockingRequestHandler(RequestHandler):
+    """
+    Decorates the :py:class:`RequestHandler` to allow responding synchronously.
+
+    This class should only be instantiated inside the implementation of the :py:class:`BlockingHttpServer`.
+    """
+
+    def __init__(self, request: Request, response_queue: Queue):
+        super().__init__(None)
```

Comment:
here, mypy will complain about passing a None as the matcher object. And it will also break Liskov I think.

So I have two ideas.
1. Don't subclass from RequestHandler, copy&paste the method definitions to here, eg to have the same look, and don't call super() at all, implement everything here. Don't know if it would look terrible, probably it would but but we can give it a try.

2.  Separate the `RequestHandler` in `httpserver.py` into two: `RequestHandlerBase`, defining some of the `respond_*` methods, tipically those which are common between `BlockingRequestHandler` and `RequestHandler`. As `RequestHandlerBase` would not define the `__init__`, this problem here would be solved.

If you want, or something is not clear, I can put together an example for this.

Reply:
I gave the matcher too, because it was available.

Reply:
Together with providing the respond_with_handler, it make sense now to pass the request.

Reply:
Yes, for example the `expect_*` methods are not used in your case.

Currently the code works, so I'm ok with merging, and then we can refactor it (before I release a new version) in a separate PR, if you want, and if that makes the situation better.

I'm also happy to do the refactoring if you are not motivated to do that (I don't want to be picky).

So, please let me know if you want a separate PR and/or want me to do the refactoring.

Reply:
I have created a refactor for the requesthandler, and pushed in the commit 9c90045709ea6ed7778c64b944b3025694d2eb79.
Feel free to cherry-pick it to your branch.

Reply:
I try to refactor the `HTTPServer` and the `RequestHandler` classes in this pull request.
I have a bit different idea that I will show in an other commit.

Reply:
Request handlers are refactored. Commits need reorder and some of the squashing.

Continuing with the server...

Reply:
Done.